server/block/explosion.go

```diff
@@ -0,0 +1,191 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/cube/trace"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math"
+	"math/rand"
+	"time"
+)
+
+// ExplosionConfig is the configuration for an explosion. The world, position, size, sound, particle, and more can all
+// be configured through this configuration.
+type ExplosionConfig struct {
+	// World is the world in which the explosion should take place.
+	World *world.World
+	// Pos is the center position of the explosion.
+	Pos mgl64.Vec3
+	// Size is the size of the explosion, it is effectively the radius which entities/blocks will be affected within.
+	Size float64
+	// RandSource is the source to use for the explosion "randomness".
+	RandSource rand.Source
+	// SpawnFire will cause the explosion to randomly start fires in 1/3 of all destroyed air blocks that are
+	// above opaque blocks.
+	SpawnFire bool
+
+	// Sound is the sound to play when the explosion is created. If set to nil, this will default to the sound of a
+	// regular explosion.
+	Sound world.Sound
+	// Particle is the particle to spawn when the explosion is created. If set to nil, this will default to the particle
+	// of a regular huge explosion.
+	Particle world.Particle
+}
+
+// ExplodableEntity represents an entity that can be exploded.
+type ExplodableEntity interface {
+	// Explode is called when an explosion occurs. The entity can then react to the explosion using the configuration
+	// and impact provided.
+	Explode(c ExplosionConfig, impact float64)
+}
+
+// Explodable represents a block that can be exploded.
+type Explodable interface {
+	// Explode is called when an explosion occurs. The block can react to the explosion using the configuration passed.
+	Explode(pos cube.Pos, c ExplosionConfig)
+}
+
+// rays ...
+var rays = make([]mgl64.Vec3, 0, 1352)
+
+// init ...
+func init() {
+	for x := 0.0; x < 16; x++ {
+		for y := 0.0; y < 16; y++ {
+			for z := 0.0; z < 16; z++ {
+				if x != 0 && x != 15 && y != 0 && y != 15 && z != 0 && z != 15 {
+					continue
+				}
+				rays = append(rays, mgl64.Vec3{x/15*2 - 1, y/15*2 - 1, z/15*2 - 1}.Normalize().Mul(0.3))
+			}
+		}
+	}
+}
+
+// Do performs the explosion as specified by the configuration.
+func (c ExplosionConfig) Do() {
```

Comment:
Would probably rename this to `Explode()` because it doesn't necessarily have some sort of `explosion.` prefix.

Reply:
Renamed.